### PR TITLE
feat: 6363-docs-initial-structure

### PR DESCRIPTION
**Pull request by motylp**

### CODE REVIEW

File: docs/workloads/common/data/pysparkle/pysparkle_data_quality.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# PySparkle Data Quality
```
Comment (motylp) [inline docs/workloads/common/data/pysparkle/pysparkle_data_quality.md:1]: 1:1 copy of PySparkle DQ README.

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md [hunk 2]
```diff
@@ -0,0 +1,107 @@
+# PySparkle
```
Comment (motylp) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:1]: 1:1 copy of PySparkle main README.md.

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Silver processing
+
+Silver processing represents the 'Bronze to Silver' layer transformations. It accepts raw data
+(CSV files) as an input and converts it into a more structured delta format.
```
Comment (iancowleyamido) [inline docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md:4]: i.e. CSV files - won't always be CSV files.

File: README.md [hunk 4]
```diff
@@ -60,29 +73,30 @@ A Makefile has been created to assist with setting up the development environmen
...
 ```
 
 ### Running unit tests
 
-In order to run unit tests run the following command:
+In order to run unit tests for the Ingest pipeline run the following command:
 
 ```bash
-make test
+make test_ingest
```
Comment (iancowleyamido) [inline README.md]: As discussed - throughout this change, I believe the testing should be at the root level, not per area i.e. not for ingest?

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+Azure Data Lake Storage (ADLS) has been chosen as both the source and destination system for our
```
Comment (iancowleyamido) [inline docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md:5]: (ADLS Gen2)

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+The naming convention for the data pipelines originates in Databricks' Medallion Architecture, which
+emphasizes structured data transformation layers. The key components of PySparkle include:
+- **Silver processing**: This represents the 'Bronze to Silver' layer transformations. It accepts
+raw data (CSV files) as an input and converts it into a more structured delta format.
+- **Gold processing**: This embodies the 'Silver to Gold' layer transformations. It performs
+advanced transformations, creating optimized, aggregated data sets for downstream consumption.
+- **Data Quality**: PySparkle performs data quality checks according to user-provided
+configurations, ensuring the reliability and accuracy of the data.
+
+We have chosen Azure Data Lake Storage (ADLS) as both the source and destination systems for our
```
Comment (iancowleyamido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:23]: (ADLS Gen2)

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+data processing due to its scalability, security features, and seamless integration with our data
+processing workflows.
+
+## PySparkle Usage
+
+> ℹ️ PySparkle Silver processing requires the following environment variables to be set
+> to access Azure Data Lake Storage (ADLS):
```
Comment (iancowleyamido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:30]: (ADLS Gen2)

File: README.md (hunk 4 above)
Comment (motylp) [reply README.md]: sure, updated

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md (hunk 7 above)
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md]: done

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md (hunk 5 above)
Comment (motylp) [reply docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md]: done

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md (hunk 3 above)
Comment (motylp) [reply docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md]: done

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
+# Silver processing
+
+Silver processing represents the 'Bronze to Silver' layer transformations. It accepts raw data
+(e.g. CSV files) as an input and converts it into a more structured delta format.
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md:18]: Maybe need to discuss this... Bronze data may already be in Parquet format (if ADF ingest loads it in this way). Further bespoke transformations will often need to happen at this layer too.

File: docs/workloads/azure/data/intro_data_azure.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+
+# Stacks – Azure Data Platform
+
+## Overview
+
+Housing an Azure data platform template, [stacks-azure-data](https://github.com/amido/stacks-azure-data)
+repository leverages **Azure Data Factory** for data ingestion and orchestration of data processing
+using **Databricks**. It also employs **Azure Data Lake Storage Gen2** for data lake storage.
+The solution's data workload naming convention originates from Databricks' Medallion Architecture,
+a system emphasizing structured data transformation layers. Key elements of the solution include:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/intro_data_azure.md:25]: We can match the wording in the main README if we're happy with it.

File: docs/workloads/azure/data/repository_data_azure.md [hunk 10]
```diff
@@ -0,0 +1,29 @@
+# Repository structure
+
+Navigation around [stacks-azure-data](https://github.com/amido/stacks-azure-data) repository:
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/repository_data_azure.md:16]:
```suggestion
Structure of [stacks-azure-data](https://github.com/amido/stacks-azure-data) repository:
```

File: docs/workloads/common/data/pysparkle/pysparkle_data_quality.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+                }
+            ]
+        }
+    ]
+}
+```
+
+### Using environment variables in configuration files
+It is possible to use environment variables in a configuration file for Data Quality.
+Placeholders in the form of {ENV_VAR_NAME} will be replaced with the corresponding environment
```
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_data_quality.md:96]:
```suggestion
Placeholders in the form of `{ENV_VAR_NAME}` will be replaced with the corresponding environment
```

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+on Kaggle, which contains metadata and ratings for 45,000 movies released on or before July 2017.
+
+**The Movies Dataset** contains a number of CSV files and includes values of different types,
+including nested datatypes and arrays, which give us enough complexity for transformations.
+
+## Key components
+The naming convention for the data pipelines originates in Databricks' Medallion Architecture, which
+emphasizes structured data transformation layers. The key components of PySparkle include:
+- **Silver processing**: This represents the 'Bronze to Silver' layer transformations. It accepts
+raw data (CSV files) as an input and converts it into a more structured delta format.
```
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:32]: As above, agree and align on this description of what Silver processing is doing

File: Makefile [hunk 13]
```diff
@@ -1,15 +1,15 @@
...
+setup_dev_environment:
 	echo "PYTHONPATH=." > .env
 	poetry self add poetry-dotenv-plugin
 	poetry shell
 	poetry update
 	pre-commit install
 
 test:
-	python -m pytest tests/unit/
+	python -m pytest ingest/jobs/Ingest_AzureSql_Example/tests/unit
```
[the remark this answers is not in the document]
Comment (motylp) [reply Makefile:9]: I did that at first, but @iancowleyamido pointed out, that noone will run ingest tests separately. So as soon as we have more tests, we will add them as separate commands within one `make test` command

File: Makefile [hunk 14]
```diff
@@ -1,15 +1,15 @@
...
 
 test_e2e:
-	behave tests/end_to_end/features/azure_data_ingest.feature
+	behave ingest/jobs/Ingest_AzureSql_Example/tests/end_to_end/features/azure_data_ingest.feature
```
[the remark this answers is not in the document]
Comment (motylp) [reply Makefile:12]: same as above :)

File: docs/workloads/azure/data/components/etl_pipelines/data_quality_azure.md [hunk 15]
```diff
@@ -0,0 +1,27 @@
+# Data Quality
+
+[stacks-azure-data](https://github.com/amido/stacks-azure-data) ETL pipelines in Azure Data Factory
+can be deployed with an extra step, 'Data Quality', which validates that the outputs of the pipeline
+meet the specified requirements. You can express these requirements in a simple, human-readable
+language. It allows you to assert certain expectations about your data, which can help catch any
+discrepancies, anomalies, or errors in your data as early in the pipeline as possible.
```
[the remark this answers is not in the document]
Comment (motylp) [reply docs/workloads/azure/data/components/etl_pipelines/data_quality_azure.md:19]: Why? I am not a fan of such long lines - not everywhere line wrapping is visible.

File: docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md (hunk 8 above)
Comment (motylp) [reply docs/workloads/azure/data/components/etl_pipelines/silver_data_azure.md:18]: Right now the documentation reflects exactly what our code currently does. I'm open to extending the scope of silver layer, but I'm not sure what are the requirements.

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md (hunk 12 above)
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:32]: Sure, but right now it's doing exactly that.

File: docs/workloads/azure/data/components/etl_pipelines/data_quality_azure.md (hunk 15 above)
[the remark this answers is not in the document]
Comment (motylp) [reply docs/workloads/azure/data/components/etl_pipelines/data_quality_azure.md:19]: Ok, sure, will look into that and update while keeping the line breaks :)

File: docs/workloads/common/data/pysparkle/pysparkle_data_quality.md [hunk 16]
```diff
@@ -0,0 +1,88 @@
...
+
+## Usage
+
+```bash
+pysparkle data-quality --help
+pysparkle data-quality --config-path "data_quality/silver_dq.json"
+```
+
+`config-path` is a Path to a JSON config inside an Azure Blob container. The container name
+for storing configurations is `config` and it is defined in the configuration set in the pysparkle
```
[the remark this answers is not in the document]
Comment (motylp) [reply docs/workloads/common/data/pysparkle/pysparkle_data_quality.md:28]: Name of the package is actually pysparkle (like all packages), so I'd keep it in lowercase.

**Review by iancowleyamido — APPROVED** [review]

File: docs/workloads/azure/data/intro_data_azure.md (hunk 9 above)
Comment (motylp) [reply docs/workloads/azure/data/intro_data_azure.md]: done

File: docs/workloads/common/data/pysparkle/pysparkle_quickstart.md [hunk 17]
```diff
@@ -0,0 +1,122 @@
...
+```
+
+## Prerequisites
+Spark runs on Java 8/11/17, Java 8 prior to version 8u362 support is deprecated
+as of Spark 3.4.0. For details see: https://spark.apache.org/docs/latest/.
+
+## Local execution
+The current setup of PySparkle doesn't include the required libraries to connect
+to Azure Data Lake Storage (they are pre-installed and configured in Azure Databricks
+environment. To run the application locally, appropriate jar files would have to be
```
Comment (adurkan-amido) [inline docs/workloads/common/data/pysparkle/pysparkle_quickstart.md:79]:
```suggestion
environment). To run the application locally, appropriate jar files would have to be
```

File: docs/workloads/azure/data/components/etl_pipelines/ingest_data_azure.md [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+            "delta_date_column": "ModifiedDate",  # (SQL sources only, delta load) Date column to use for filtering the date range
+            "delta_upsert_key": "SalesOrderID"    # (SQL sources only, delta load) Primary key for determining updated columns in a delta load
+        }
+    ]
+}
+```
+
+These configuration files will be loaded in for a given data source every time an ingestion pipeline
+is triggered in Data Factory, and all entities will be ingested. To disable a particular ingest
+source or entitiy without removing it, you can set `"enabled": false` – these will be ignored by
```
Comment (adurkan-amido) [inline docs/workloads/azure/data/components/etl_pipelines/ingest_data_azure.md:67]:
```suggestion
source or entity without removing it, you can set `"enabled": false` – these will be ignored by
```

File: pysparkle/README.md [hunk 19]
```diff
@@ -1,14 +1,40 @@
...
+**The Movies Dataset** contains a number of CSV files and includes values of different types,
+including nested datatypes and arrays, which give us enough complexity for transformations.
+
+## Key components
+The naming convention for the data pipelines originates in Databricks' Medallion Architecture, which
+emphasizes structured data transformation layers. The key components of PySparkle include:
+- **Silver processing**: This represents the 'Bronze to Silver' layer transformations. It accepts
+raw data (CSV files) as an input and converts it into a more structured delta format.
+- **Gold processing**: This embodies the 'Silver to Gold' layer transformations. It performs
+advanced transformations, creating optimized, aggregated data sets for downstream consumption.
```
Comment (adurkan-amido) [inline pysparkle/README.md:19]:
```suggestion
advanced transformations, creating optimised, aggregated data sets for downstream consumption.
```

**Review by adurkan-amido — COMMENTED** [review]
Looks good, just spell check and sticking with British English spellings

**Pull request merged**